url/url.pyx

```diff
@@ -0,0 +1,265 @@
+# cython: linetrace=True
+# distutils: define_macros=CYTHON_TRACE=1
+
+from cython.operator import dereference
+
+from publicsuffix import PublicSuffixList
+psl = PublicSuffixList()
+
+def ParseMethod(cls, s, encoding='utf-8'):
+    if isinstance(s, bytes):
+        if encoding == 'utf-8':
+            return cls(s)
+        else:
+            return cls(s.decode(encoding).encode('utf-8'))
+    else:
+        return cls(s.encode('utf-8'))
+
+
+cdef class URL:
+    '''Wrapper around url-cpp.'''
+
+    cdef Url *ptr
+
+    parse = classmethod(ParseMethod)
+
+    def __cinit__(self, const string& s):
+        self.ptr = new Url(s)
+
+    def __dealloc__(self):
+        del self.ptr
+
+    property scheme:
+        def __get__(self):
+            return self.ptr.scheme()
+        def __set__(self, s):
+            self.ptr.setScheme(s)
+
+    property host:
+        def __get__(self):
+            return self.ptr.host()
+        def __set__(self, s):
+            self.ptr.setHost(s)
+
+    property port:
+        def __get__(self):
+            return self.ptr.port()
+        def __set__(self, i):
+            self.ptr.setPort(i)
+
+    property path:
+        def __get__(self):
+            return self.ptr.path()
+        def __set__(self, s):
+            self.ptr.setPath(s)
+
+    property params:
+        def __get__(self):
+            return self.ptr.params()
+        def __set__(self, s):
+            self.ptr.setParams(s)
+
+    property query:
+        def __get__(self):
+            return self.ptr.query()
+        def __set__(self, s):
+            self.ptr.setQuery(s)
+
+    property fragment:
+        def __get__(self):
+            return self.ptr.fragment()
+        def __set__(self, s):
+            self.ptr.setFragment(s)
+
+    property userinfo:
+        def __get__(self):
+            return self.ptr.userinfo()
+        def __set__(self, s):
+            self.ptr.setUserinfo(s)
+
+    def copy(self):
+        '''Return a new instance of an identical URL.'''
+        new = URL('')
+        new.ptr.setScheme(self.ptr.scheme())
```

Comment:
Seems like maybe you should add a copy constructor to `url-cpp` instead of implementing this in Cython.